src/Peering/Peering/PeerAsn/SetAzurePeerAsnCommand.cs

```diff
@@ -0,0 +1,182 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+namespace Microsoft.Azure.PowerShell.Cmdlets.Peering.PeerAsn
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Management.Automation;
+
+    using Microsoft.Azure.Commands.Peering.Properties;
+    using Microsoft.Azure.Management.Peering;
+    using Microsoft.Azure.Management.Peering.Models;
+    using Microsoft.Azure.PowerShell.Cmdlets.Peering.Common;
+    using Microsoft.Azure.PowerShell.Cmdlets.Peering.Models;
+    using Microsoft.Rest.Azure;
+
+    using Newtonsoft.Json;
+
+    /// <summary>
+    ///     New Azure InputObject Command-let
+    /// </summary>
+    [Cmdlet(
+        VerbsCommon.Set,
+        "AzPeerAsn",
+        DefaultParameterSetName = Constants.ParameterSetNameByName,
+        SupportsShouldProcess = true)]
+    [OutputType(typeof(PSPeerAsn))]
+    public class SetAzurePeerAsn : PeeringBaseCmdlet
+    {
+        /// <summary>
+        /// Gets or sets the legacy InputObject.
+        /// </summary>
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipeline = true,
+            HelpMessage = Constants.PeerAsnHelp,
+            ParameterSetName = Constants.ParameterSetNameDefault,
+            DontShow = true)]
```

Comment:
@rileymckenna it looks like t here are a few parameters that have this `DontShow` property -- any reason they have this?

Reply:
no it was a copy and paste from i think CDN. i just thought it was the convention. i removed them.